modules/frontend/frontend.go

```diff
@@ -55,19 +56,19 @@ func New(cfg Config, next http.RoundTripper, o overrides.Interface, reader tempo
 		return nil, fmt.Errorf("query backend after should be less than or equal to query ingester until")
 	}
 
-	retryWare := newRetryWare(cfg.MaxRetries, registerer)
-
-	// cache
-	searchCache := newFrontendCache(cacheProvider, cache.RoleFrontendSearch, logger)
+	retryWare := pipeline.NewRetryWare(cfg.MaxRetries, registerer)
+	cacheWare := pipeline.NewCachingWare(cacheProvider, cache.RoleFrontendSearch, logger)
+	statusCodeWare := pipeline.NewStatusCodeAdjustWare()
 
 	// inject multi-tenant middleware in multi-tenant routes
 	traceByIDMiddleware := MergeMiddlewares(
 		newMultiTenantMiddleware(cfg, combiner.NewTraceByID, logger),
 		newTraceByIDMiddleware(cfg, o, logger), retryWare)
 
-	searchMiddleware := MergeMiddlewares(
-		newMultiTenantMiddleware(cfg, combiner.NewSearch, logger),
-		newSearchMiddleware(cfg, o, reader, searchCache, logger), retryWare)
+	searchPipeline := pipeline.Build(
```

Comment:
Double checking my understanding:  Search is the only API moved to the new pipeline?